src/kernel/arch/x86/paging.zig

```diff
@@ -124,52 +317,121 @@ pub fn init(mem_profile: *const MemProfile, allocator: *std.mem.Allocator) void
     const p_start = std.mem.alignBackward(@ptrToInt(mem_profile.physaddr_start), PAGE_SIZE);
     const p_end = std.mem.alignForward(@ptrToInt(mem_profile.physaddr_end) + mem_profile.fixed_alloc_size, PAGE_SIZE);
 
-    var tmp = allocator.alignedAlloc(Directory, PAGE_SIZE_4KB, 1) catch unreachable;
+    var tmp = allocator.alignedAlloc(Directory, PAGE_SIZE_4KB, 1) catch panic.panicFmt(@errorReturnTrace(), "Failed to allocate page directory");
     var kernel_directory = @ptrCast(*Directory, tmp.ptr);
     @memset(@ptrCast([*]u8, kernel_directory), 0, @sizeOf(Directory));
 
-    mapDir(kernel_directory, p_start, p_end, v_start, v_end, allocator) catch unreachable;
-    const dir_physaddr = @ptrToInt(kernel_directory) - @ptrToInt(&KERNEL_ADDR_OFFSET);
+    // Map in kernel
+    mapDir(kernel_directory, p_start, p_end, v_start, v_end, allocator) catch panic.panicFmt(@errorReturnTrace(), "Failed to map kernel directory");
+    const tty_addr = tty.getVideoBufferAddress();
+    // If the previous mappping space didn't cover the tty buffer, do so now
+    if (v_start > tty_addr or v_end <= tty_addr) {
+        const tty_phys = virtToPhys(tty_addr);
+        const tty_buff_size = 32 * 1024;
```

Comment:
Is this correct? The TTY buffer would be 80x25 characters x2 for colour per character. So would be 80x25x2=4000

Reply:
I read somewhere that it was 32KB but your calculation makes more sense.

Reply:
Page faults happen when using 4Kb so 32Kb it is :p

Reply:
I think the video buffer is 128KB (0x000A0000 - 0x000BFFFF) but I only use a bit of it from (0x000B0000 - 0x000BFFFF) which is 32KB.